fortran/micm.F90

```diff
@@ -25,13 +26,22 @@ module musica_micm
       real(c_double)     :: final_time_ = 0._c_double
    end type solver_stats_t_c
 
+   ! We could use Fortran 2023 enum type feature if Fortran 2023 is supported
+   ! https://fortran-lang.discourse.group/t/enumerator-type-in-bind-c-derived-type-best-practice/5947/2
+   enum, bind(c)
+      enumerator :: Rosenbrock              = 1
+      enumerator :: RosenbrockStandardOrder = 2
+   end enum
```

Comment:
As long as our CI tests pass then I'm fine with this. The only potential issue maybe is the NAG compiler and if it supports this and if we need to compile with NAG ever.

But I like this as it is

Reply:
Jesse mentioned that it's probably not yet to use Fortran 2023 features but I'll leave the comment so we can revisit later.

Reply:
I like this too